systemtest/src/main/java/io/strimzi/systemtest/performance/report/parser/PerformanceMetricsParser.java

```diff
@@ -0,0 +1,392 @@
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+// TODO: Each component should have it's own parser I would say (different metrics)
+// TODO: We should have some base/common parser on top of the hierarchy to share common performance metrics
+/**
+ * Parses and displays performance metrics from experiment results in {@code TestUtils.USER_PATH + "/target/performance}
+ * directory
+ * <p>
+ * This class is designed to read performance metrics stored in files, which are generated
+ * from performance tests. It categorizes metrics by use case and experiment, processes them to
+ * calculate various statistical measures, and outputs formatted results. The parser can
+ * differentiate between running in a test environment and a standalone application mode,
+ * adjusting file paths accordingly. Metrics include operation durations, system and JVM
+ * performance, and custom metrics defined in {@code PerformanceConstants}.
+ * <p>
+ * Usage:
+ * <ul>
+ *     <li>To run as a standalone application, provide the directory path containing
+ *     performance metrics files as a command-line argument. If no directory is specified,
+ *     the parser attempts to find the latest directory under the default path.</li>
+ *     <li>In a test environment, ensure the {@code sun.java.command} system property
+ *     contains "JUnitStarter" to adjust the base path for metrics files.</li>
+ * </ul>
+ *
+ * Key Methods:
+ * <ul>
+ *     <li>{@link #isRunningInTest()} - Checks if the application is running in a test environment.</li>
+ *     <li>{@link #main(String[])} - Entry point for parsing and displaying performance metrics.</li>
+ *     <li>{@link #processFile(File, ExperimentMetrics)} - Processes individual metrics files to extract metrics.</li>
+ *     <li>{@link #showValuesOfExperiments()} - Displays formatted metrics for all experiments and use cases.</li>
+ * </ul>
+ */
+public class PerformanceMetricsParser {
+
+    private static final Map<String, List<ExperimentMetrics>> USE_CASE_EXPERIMENTS = new HashMap<>();
+    private static final Logger LOGGER = LogManager.getLogger(PerformanceMetricsParser.class);
+
+    public static boolean isRunningInTest() {
+        for (StackTraceElement element : Thread.currentThread().getStackTrace()) {
+            if (element.getClassName().startsWith("org.junit.")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @SuppressWarnings({"checkstyle:CyclomaticComplexity"})
+    public static void main(String[] args) throws IOException {
+
+        LOGGER.info("Using user.dir: {}", TestUtils.USER_PATH);
+
+        String parentPath;
+        // resolve path for TestingFarm
+        if (System.getenv().containsKey("TMT_PLAN_DATA")) {
+            parentPath = System.getenv().get("TMT_PLAN_DATA") + "/../discover/default-0/tests/systemtest/target/performance";
```

Comment:
Hmm
Is it possible to pass this via the `PERFORMANCE_DIR` (or whatever) env variable? I guess that having it hardcoded in the code can cause issues. I guess this applies for other paths.